File: apps/Applications/ios/Local For Vocal/Local For Vocal/Sources/Views/Components/SDUI/PoweredByRowSDUI.swift
import SwiftUI

struct PoweredByRowSDUI: View {
    let component: SDUIComponent

    var body: some View {
        let title = component.prop(for: "title") as String? ?? "POWERED BY"
        let brands = component.decodeItems(for: "brands", as: [PoweredByBrand].self)

        VStack(spacing: 8) {
            Text(title)
                .font(.system(size: 10, weight: .bold))
                .foregroundColor(.white.opacity(0.7))
                .tracking(2)

            HStack(spacing: 20) {
                ForEach(brands, id: \.name) { brand in
                    if let logoUrl = brand.logoUrl {
                        // Image Logo
                        CachedAsyncImage(url: URL(string: logoUrl)) { image in
                            image.resizable().scaledToFit()
                        } placeholder: {
                            Color.clear
                        }
                        .frame(height: 24)
                        .padding(.horizontal, 8)
                        .padding(.vertical, 4)
                        .background(Color.white.opacity(0.1))
                        .cornerRadius(4)
                    } else {
                        // Text Logo Fallback
                        Text(brand.name)
                            .font(.system(size: 14, weight: .black))
                            .foregroundColor(Color(hex: brand.textColor ?? "#FFFFFF"))
                            .padding(.horizontal, 8)
                            .padding(.vertical, 4)
                            .background(Color.white.opacity(0.1))
                            .cornerRadius(4)
                    }
                }
            }
        }
        .padding(.vertical, 8)
    }
}

struct PoweredByBrand: Decodable {
    let name: String
    let logoUrl: String?
    let textColor: String?
}

File: apps/Applications/android/app/src/main/java/com/ecommerceearn/app/ui/components/PoweredByRowSDUI.kt
package com.ecommerceearn.app.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import coil.compose.AsyncImage
import com.ecommerceearn.app.data.model.SDUIComponent

@Composable
fun PoweredByRowSDUI(component: SDUIComponent) {
    val props = component.props
    val title = props?.get("title")?.toString() ?: "POWERED BY"
    
    // Attempt to extract brands list 
    val brandsArray = props?.get("brands") as? List<*> ?: emptyList<Any>()

    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 8.dp),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        // Title with transparency mapping iOS tracking
        Text(
            text = title.uppercase(),
            fontSize = 10.sp,
            fontWeight = FontWeight.Bold,
            color = Color.White.copy(alpha = 0.7f),
            letterSpacing = 2.sp
        )

        Spacer(modifier = Modifier.height(8.dp))

        Row(
            horizontalArrangement = Arrangement.spacedBy(20.dp, Alignment.CenterHorizontally),
            verticalAlignment = Alignment.CenterVertically
        ) {
            brandsArray.forEach { brandMapRaw ->
                val brandMap = brandMapRaw as? Map<*, *> ?: return@forEach
                val name = brandMap["name"]?.toString() ?: ""
                val logoUrl = brandMap["logoUrl"]?.toString()
                val textColorHex = brandMap["textColor"]?.toString() ?: "#FFFFFF"
                
                val displayColor = try {
                    Color(android.graphics.Color.parseColor(textColorHex))
                } catch (e: Exception) {
                    Color.White
                }

                if (!logoUrl.isNullOrEmpty()) {
                    AsyncImage(
                        model = logoUrl,
                        contentDescription = name,
                        modifier = Modifier
                            .height(24.dp)
                            .background(Color.White.copy(alpha = 0.1f), RoundedCornerShape(4.dp))
                            .padding(horizontal = 8.dp, vertical = 4.dp)
                    )
                } else {
                    Text(
                        text = name,
                        fontSize = 14.sp,
                        fontWeight = FontWeight.Black,
                        color = displayColor,
                        modifier = Modifier
                            .background(Color.White.copy(alpha = 0.1f), RoundedCornerShape(4.dp))
                            .padding(horizontal = 8.dp, vertical = 4.dp)
                    )
                }
            }
        }
    }
}
